internal/api/v1beta1/domain_test.go

```diff
@@ -0,0 +1,617 @@
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/raystack/frontier/core/authenticate"
+	"github.com/raystack/frontier/core/domain"
+	"github.com/raystack/frontier/core/organization"
+	"github.com/raystack/frontier/internal/api/v1beta1/mocks"
+	frontierv1beta1 "github.com/raystack/frontier/proto/v1beta1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestHandler_CreateOrganizationDomain(t *testing.T) {
+
+	tests := []struct {
+		name    string
+		setup   func(m *mocks.DomainService)
+		req     *frontierv1beta1.CreateOrganizationDomainRequest
+		want    *frontierv1beta1.CreateOrganizationDomainResponse
+		wantErr error
+	}{
+		{name: "should create Organization Domain on success",
+			setup: func(m *mocks.DomainService) {
+				m.EXPECT().Create(mock.AnythingOfType("*context.emptyCtx"), domain.Domain{
+					OrgID: "org_id",
+					Name:  "domain_name",
+				}).Return(domain.Domain{
+					ID:    "some_id",
+					Name:  "domain_name",
```

Comment:
We can use a sample name like `example.com` as it will actually be a domain name.